### PR TITLE
Add user controller

### PR DESCRIPTION
**Pull request by BrainMaestro**
Completes User in #62

### CODE REVIEW

**Comment by BrainMaestro** [comment]
Approved

File: app/Http/Requests/UserCreateRequest.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'name' => 'required|min:3',
+            'email' => 'required|email|regex:/@aun.edu.ng$/',
```
Comment (SkaeX) [inline app/Http/Requests/UserCreateRequest.php]: There are users who don't use AUN emails.

File: app/Http/Requests/UserUpdateRequest.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'name' => 'min:3',
+            'email' => 'required_with:password|email|regex:/@aun.edu.ng$/',
```
Comment (SkaeX) [inline app/Http/Requests/UserUpdateRequest.php]: I have complained about this.

File: app/Http/Kernel.php [hunk 3]
```diff
@@ -17,7 +17,7 @@ class Kernel extends HttpKernel
         \Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse::class,
         \Illuminate\Session\Middleware\StartSession::class,
         \Illuminate\View\Middleware\ShareErrorsFromSession::class,
-        \Fce\Http\Middleware\VerifyCsrfToken::class,
+//        \Fce\Http\Middleware\VerifyCsrfToken::class,
```
Comment (SkaeX) [inline app/Http/Kernel.php]: Why?

File: app/Repositories/Repository.php [hunk 4]
```diff
@@ -122,6 +122,10 @@ protected function findBy(array $params, $limit = 15, $page = 1, array $columns
      */
     protected function update($id, array $attributes)
     {
+        if (count($attributes) == 0) {
```
Comment (SkaeX) [inline app/Repositories/Repository.php]: Will there be any need to check the `$id` at the controller?

Comment (BrainMaestro) [inline app/Repositories/Repository.php]: I don't understand.


Sent from Outlook Mobile






On Sun, Mar 6, 2016 at 4:02 AM -0800, "SkaeX" <notifications@github.com> wrote:





> @@ -122,6 +122,10 @@ protected function findBy(array $params, $limit = 15, $page = 1, array $columns
>       */
>      protected function update($id, array $attributes)
>      {
> +        if (count($attributes) == 0) {

Will there be any need to check the `$id` at the controller?

---
Reply to this email directly or view it on GitHub:
https://github.com/BrainMaestro/FCE/pull/72/files#r55137961

File: app/Http/Requests/UserCreateRequest.php (hunk 1 above)
Comment (BrainMaestro) [inline app/Http/Requests/UserCreateRequest.php]: Should it really be allowed? Every instructor should have an AUN email.


Can you confirm from the OIRE.


Sent from Outlook Mobile






On Sun, Mar 6, 2016 at 3:57 AM -0800, "SkaeX" <notifications@github.com> wrote:





> +     */
> +    public function authorize()
> +    {
> +        return true;
> +    }
> +
> +    /**
> +     * Get the validation rules that apply to the request.
> +     *
> +     * @return array
> +     */
> +    public function rules()
> +    {
> +        return [
> +            'name' => 'required|min:3',
> +            'email' => 'required|email|regex:/@aun.edu.ng$/',

There are users who don't use AUN emails.

---
Reply to this email directly or view it on GitHub:
https://github.com/BrainMaestro/FCE/pull/72/files#r55137853

File: app/Repositories/Repository.php (hunk 4 above)
Comment (SkaeX) [inline app/Repositories/Repository.php]: Extra check on `$id`

File: app/Http/Kernel.php (hunk 3 above)
Comment (BrainMaestro) [inline app/Http/Kernel.php]: It was causing a token mismatch issue when testing with postman. We'll add it back when the front end is built.


Sent from Outlook Mobile






On Sun, Mar 6, 2016 at 4:01 AM -0800, "SkaeX" <notifications@github.com> wrote:





> @@ -17,7 +17,7 @@ class Kernel extends HttpKernel
>          \Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse::class,
>          \Illuminate\Session\Middleware\StartSession::class,
>          \Illuminate\View\Middleware\ShareErrorsFromSession::class,
> -        \Fce\Http\Middleware\VerifyCsrfToken::class,
> +//        \Fce\Http\Middleware\VerifyCsrfToken::class,

Why?


---
Reply to this email directly or view it on GitHub:
https://github.com/BrainMaestro/FCE/pull/72/files#r55137915

File: app/Repositories/Repository.php (hunk 4 above)
Comment (BrainMaestro) [inline app/Repositories/Repository.php]: What kind of check exactly


Sent from Outlook Mobile






On Sun, Mar 6, 2016 at 4:46 AM -0800, "SkaeX" <notifications@github.com> wrote:





> @@ -122,6 +122,10 @@ protected function findBy(array $params, $limit = 15, $page = 1, array $columns
>       */
>      protected function update($id, array $attributes)
>      {
> +        if (count($attributes) == 0) {

Extra check on `$id`


---
Reply to this email directly or view it on GitHub:
https://github.com/BrainMaestro/FCE/pull/72/files#r55138576

File: app/Http/Requests/UserCreateRequest.php (hunk 1 above)
Comment (SkaeX) [inline app/Http/Requests/UserCreateRequest.php]: Things are getting really sloppy. I remember an instructor `webertrand@gmail.com`. However, I personally think this is a case of "This is the rule" like we had with the unique `crn`. I will confirm with Esi/Fidelis though.

File: app/Repositories/Repository.php (hunk 4 above)
Comment (SkaeX) [inline app/Repositories/Repository.php]: Nevermind.

File: app/Http/Kernel.php (hunk 3 above)
Comment (SkaeX) [inline app/Http/Kernel.php]: Alright.

**Comment by SkaeX** [comment]
Approved.

File: app/Http/Requests/UserCreateRequest.php (hunk 1 above)
Comment (BrainMaestro) [inline app/Http/Requests/UserCreateRequest.php]: When you do let me. It's a minor thing.


Sent from Outlook Mobile






On Sun, Mar 6, 2016 at 4:54 AM -0800, "SkaeX" <notifications@github.com> wrote:





> +     */
> +    public function authorize()
> +    {
> +        return true;
> +    }
> +
> +    /**
> +     * Get the validation rules that apply to the request.
> +     *
> +     * @return array
> +     */
> +    public function rules()
> +    {
> +        return [
> +            'name' => 'required|min:3',
> +            'email' => 'required|email|regex:/@aun.edu.ng$/',

Things are getting really sloppy. I remember an instructor `webertrand@gmail.com`. However, I personally think this is a case of "This is the rule" like we had with the unique `crn`. I will confirm with Esi/Fidelis though.

---
Reply to this email directly or view it on GitHub:
https://github.com/BrainMaestro/FCE/pull/72/files#r55138643

File: app/Repositories/Repository.php (hunk 4 above)
Comment (tenorz007) [inline app/Repositories/Repository.php]: Why did you do this? @BrainMaestro

Comment (BrainMaestro) [inline app/Repositories/Repository.php]: I think some the update was getting an empty array in some cases. I'll remove it when all the validation is done.

**Pull request merged**